### ISSUE TITLE
Boostrap CSS resource requested in OpenLayers resource previewer is 404

### ISSUE THREAD

**Issue by gjlawran** [bcgov/ckanext-bcgov#231]
In production - resource webpages that use the OpenLayers map previewer - e.g. these records that have WMS resources:

https://catalogue.data.gov.bc.ca/dataset/tantalis-regional-districts/resource/ad41d2be-4df2-44ad-af95-
570a15781a52
https://catalogue.data.gov.bc.ca/dataset/important-fossil-areas/resource/ae7458ae-500c-4c58-96e8-ea9c40639b1d

are generating a 404 response for this resource  https://catalogue.data.gov.bc.ca/css/bootstrap.min.css

**Comment by kfishwick** [comment]
Analysis please @jrods ?

**Comment by jrods** [comment]
Looks like the bootstrap.min.css is being loaded from the ext that provides the map view, still investigating.

**Comment by jrods** [comment]
I can't reproduce, my env doesn't request the minified version of bootstrap at all, even when debug is off @ll911

**Comment by ll911** [comment]
it's ref in requests `https://catalogue.data.gov.bc.ca/dataset/tantalis-regional-districts/resource/ad41d2be-4df2-44ad-af95-570a15781a52`
line 23
```html
<link rel="stylesheet" href="/css/bootstrap.min.css" type="text/css" media="screen, projection" />
```

**Comment by jrods** [comment]
@ll911 I know, but I can't get bootstrap.min.css to be called, instead on my local env bootstrap.css is call (which is http 200), when debug is either on or off.

**Comment by ll911** [comment]
@jrods that's odd, i got 404 for all 3 env cad/cat/prod

**Comment by ll911** [comment]
it's referenced by geoview extension, did you installed the geoview extension? e.g 
```templates/dataviewer/openlayers3.html```

**Comment by jrods** [comment]
yes i have the geoview ext installed and that code block is present in my env

**Comment by ll911** [comment]
@jrods base difference, simply update that line reference for bootstrap css file to ```base/vendor/bootstrap/js/bootstrap.min.js```

**Comment by jrods** [comment]
An update on this issue, I'm able to reproduce in my env, it depends on the resource used. Anyhow, I couldn't find any file `css/bootstrap.css` or `css/bootstrap.min.css` in ckan and ckanexts that are used, that the geoview ext references. 

In this commit, those references are actually removed -> https://github.com/ckan/ckanext-geoview/commit/380278ca644363ee64356a600fdac544a09ab6c2 

But were later, the changes were reverted -> https://github.com/ckan/ckanext-geoview/commit/71f59cdb6b4cd598018f71655e772ea86289578e

I've pinged one of the ckan devs to inquire about this and see where the intended location of bootstrap.min.css is suppose to be.

**Comment by ll911** [comment]
@jrods they are here https://github.com/ckan/ckan/tree/ckan-2.5.4/ckan/public/base/vendor/bootstrap/js

**Comment by jrods** [comment]
@ll911 those are js files, not css files

**Comment by jrods** [comment]
So after my analysis, there are a few of options.

The included bootstrap source files dont have the appropriate files to create bootstrap.min.css, modification of bcgov ckan will have to be made.

The recline view ext in core ckan has a bootstrap.css file, but again, modifications have to be made to bcgov ckan.

The other option I would like to suggest, is to simply remove the reference. The view already uses style.css from its extension. Changes only apply to bcgov ckanext-geoview.

@Mbrownshoes

**Comment by ll911** [comment]
@jrods @Mbrownshoes what about using cdn instead of including the js files in the source
https://www.bootstrapcdn.com/

**Comment by Mbrownshoes** [comment]
@ll911 what do you think of @jrods suggestion? What do you think is the best solution here?

**Comment by ll911** [comment]
both css and js files are available to cdn. if bootstrap not required, yes we can remove it. but did you verify that's the case for both desktop and mobile interface?

**Comment by jrods** [comment]
I can verify that there is no difference for both desktop and mobile and the reference can be remove for bootstrap.min.css.

**Comment by jrods** [comment]
@ll911 What branch is being used in cad for ckanext-geoview?

i just merged my changes to the geoview ext, but they're not working.

**Comment by ll911** [comment]
@jrods https://gogs.data.gov.bc.ca/bcdc/deployment/src/master/cad/Jenkinsfile#L11

**Comment by jrods** [comment]
@ll911 why is the ckanext-geoview using the 0.0.12-bcgov branch as the default? it should be set to master if jenkins uses the master branch.

**Comment by ll911** [comment]
@jrods it was pinned version since we were ahead of the origin, if origin has merged our changes, we can switch back, can you confirm that?

**Comment by ll911** [comment]
@jrods you can change src `https://github.com/bcgov/ckanext-geoview` to `https://github.com/ckan/ckanext-geoview` in [Jenkinsfile](https://gogs.data.gov.bc.ca/bcdc/deployment/src/master/cad/Jenkinsfile#L11)

**Comment by jrods** [comment]
@ll911 origin will always be behind, i think it would be better to have jenkins use 0.0.12-bcgov branch instead

**Comment by ll911** [comment]
@jrods sure, you can change that part in the same place.

**Comment by jrods** [comment]
@Mbrownshoes cad has now been update with these changes, bootstrap.min.css is no longer referenced

**Comment by dkelsey** [comment]
@kfishwick @jrods @ll911  
 What is the test case?  

Is it simply:

PROD version gets the bootstrap.css error:
[https://catalogue.data.gov.bc.ca/dataset/tantalis-regional-districts/resource/ad41d2be-4df2-44ad-af95-570a15781a52](https://catalogue.data.gov.bc.ca/dataset/tantalis-regional-districts/resource/ad41d2be-4df2-44ad-af95-570a15781a52)

TEST version doesn't:
[https://cat.data.gov.bc.ca/dataset/tantalis-regional-districts/resource/f27d27e2-6344-4507-a484-f5a6fc45d376](https://cat.data.gov.bc.ca/dataset/tantalis-regional-districts/resource/f27d27e2-6344-4507-a484-f5a6fc45d376)

Can this JIRA be moved to **Verify in PROD**?

**Comment by dkelsey** [comment]
When I got to:
https://catalogue.data.gov.bc.ca/api/3/action/organization_list?organizations=%22bc-film-commission%22,%22city-of-surrey%22&all_fields=1

The Map preview appears blank:
![blank-map](https://user-images.githubusercontent.com/3906047/30765494-cf8edec2-9fa4-11e7-853b-ebe0c3db53cd.PNG)


I get the following in the Console:
```
Uncaught TypeError: Cannot read property 'layers' of undefined
    at ol2_preview.js:5198
    at success (ol2_preview.js:5043)
    at Object.runCallbacks (ol2_preview.js:1871)
    at fXMLHttpRequest.request.onreadystatechange (ol2_preview.js:1867)
    at fXMLHttpRequest.cXMLHttpRequest.dispatchEvent (ol2_preview.js:1901)
    at fReadyStateChange (ol2_preview.js:1904)
    at XMLHttpRequest._object.onreadystatechange (ol2_preview.js:1887)
```

**Comment by ll911** [comment]
doesn’t seem like there are any geojson in the resource

https://catalogue.data.gov.bc.ca/dataset?organization=bc-film-commission

**Comment by dkelsey** [comment]
Of the two links in the initial comments, the first link works fails, the second one succeeds.
Is this a different problem?

**Comment by ll911** [comment]
@dkelsey https://catalogue.data.gov.bc.ca/dataset/tantalis-regional-districts has resource problem

**Comment by ll911** [comment]
@dkelsey only second link has valid resource, first one does not have valid resource anymore, needs resource update

**Comment by dkelsey** [comment]
I'm moving this to 1.7.  Unless someone can tell if if it is fixed and can be closed.
